pandas/core/dtypes/dtypes.py

```diff
@@ -1008,27 +1008,33 @@ class IntervalDtype(PandasExtensionDtype):
     base = np.dtype("O")
     num = 103
     _metadata = ("subtype",)
-    _match = re.compile(r"(I|i)nterval\[(?P<subtype>.+)\]")
+    _match = re.compile(
+        r"(I|i)nterval\[(?P<subtype>[^,]+)(, (?P<closed>(right|left|both|neither)))?\]"
+    )
     _cache: Dict[str_type, PandasExtensionDtype] = {}
 
-    def __new__(cls, subtype=None):
+    def __new__(cls, subtype=None, closed=None):
         from pandas.core.dtypes.common import is_string_dtype, pandas_dtype
 
         if isinstance(subtype, IntervalDtype):
             return subtype
+            # TODO: what if closed is also passed?
```

Comment:
prob should raise (unless this is for pickle compat)

Reply:
will do